packages/network-controller/src/rpc-service/rpc-service-chain.test.ts

```diff
@@ -537,159 +651,1446 @@ describe('RpcServiceChain', () => {
+      const tertiaryEndpointUrl = 'https://second.endpoint';
+      nock(primaryEndpointUrl)
+        .post('/', {
+          id: 1,
+          jsonrpc: '2.0',
+          method: 'eth_chainId',
+          params: [],
+        })
+        .times(DEFAULT_MAX_CONSECUTIVE_FAILURES)
+        .reply(503);
+      nock(secondaryEndpointUrl)
+        .post('/', {
+          id: 1,
+          jsonrpc: '2.0',
+          method: 'eth_chainId',
+          params: [],
+        })
+        .times(DEFAULT_MAX_CONSECUTIVE_FAILURES)
+        .reply(503);
+      nock(tertiaryEndpointUrl)
+        .post('/', {
+          id: 1,
+          jsonrpc: '2.0',
+          method: 'eth_chainId',
+          params: [],
+        })
+        .times(DEFAULT_MAX_CONSECUTIVE_FAILURES)
+        .reply(503);
+      const expectedError = createResourceUnavailableError(503);
+      const rpcServiceChain = new RpcServiceChain([
+        {
+          fetch,
+          btoa,
+          endpointUrl: primaryEndpointUrl,
+        },
+        {
+          fetch,
+          btoa,
+          endpointUrl: secondaryEndpointUrl,
+        },
+        {
+          fetch,
+          btoa,
+          endpointUrl: tertiaryEndpointUrl,
+        },
+      ]);
+      const onServiceBreakListener = jest.fn();
+      rpcServiceChain.onServiceRetry(() => {
+        clock.next();
+      });
+      rpcServiceChain.onServiceBreak(onServiceBreakListener);
+
+      const jsonRpcRequest = {
+        id: 1,
+        jsonrpc: '2.0' as const,
+        method: 'eth_chainId',
+        params: [],
+      };
+      // Retry the first endpoint until max retries is hit.
+      await expect(rpcServiceChain.request(jsonRpcRequest)).rejects.toThrow(
+        expectedError,
+      );
+      // Retry the first endpoint again, until max retries is hit.
+      await expect(rpcServiceChain.request(jsonRpcRequest)).rejects.toThrow(
+        expectedError,
+      );
+      // Retry the first endpoint for a third time, until max retries is hit.
+      // The circuit will break on the last time, and the second endpoint will
+      // be hit, until max retries is hit.
+      await expect(rpcServiceChain.request(jsonRpcRequest)).rejects.toThrow(
+        expectedError,
+      );
+      // Retry the second endpoint again, until max retries is hit.
+      await expect(rpcServiceChain.request(jsonRpcRequest)).rejects.toThrow(
+        expectedError,
+      );
+      // Retry the second endpoint for a third time, until max retries is hit.
+      // The circuit will break on the last time, and the third endpoint will
+      // be hit, until max retries is hit.
+      await expect(rpcServiceChain.request(jsonRpcRequest)).rejects.toThrow(
+        expectedError,
+      );
+      // Retry the third endpoint again, until max retries is hit.
+      await expect(rpcServiceChain.request(jsonRpcRequest)).rejects.toThrow(
+        expectedError,
+      );
+      // Retry the second endpoint for a third time, until max retries is hit.
```

Comment:
This is the third, not the second:
```suggestion
      // Retry the third endpoint for a third time, until max retries is hit.
```